test/grep.js

```diff
@@ -174,3 +174,166 @@ test('the pattern looks like an option', t => {
   t.falsy(shell.error());
   t.is(result.toString(), '-v\n-vv\n');
 });
+
+//
+// Before & after contexts
```

Comment:
Can you add a few more test cases?

* `-C 0` should behave like regular grep, except it should insert line separators
* `-C -1` should trigger an error with `common.error('grep: -1: invalid context length argument')`
* `-C 100` (some number bigger than the total input file length) should just print the whole file without errors, assuming there is at least one match for the regex pattern